### PR TITLE
Update/merge conda-forge 3.7.1

### PR DESCRIPTION
**Pull request by ScottBailey**
Merged in conda-forge 3.7.1 changes, added wheel to requirements, added dev_url.
Updates Twine to 3.7.1 from 3.4.1.

License:
  https://github.com/pypa/twine/blob/main/LICENSE
  Apache 2.0 ✅ 

Changelog:
   https://github.com/pypa/twine/blob/main/docs/changelog.rst
   Mostly bugfixes, but also "Added support for core metadata version 2.2" and "Added support for Python 3.10"
 
Issues:
  https://github.com/pypa/twine/issues
  All issues seem to be code improvements or feature requests.

### CODE REVIEW

**Comment by ScottBailey** [comment]
s390x dependencies are unmet, so this doesn't built. What's the righ thing to do here?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -18,15 +19,16 @@ build:
 requirements:
   host:
     - pip
-    - python
+    - python >=3.6
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: For noarch there should be plain `python` without pinning in  host

File: recipe/meta.yaml [hunk 2]
```diff
@@ -18,15 +19,16 @@ build:
...
     - colorama >=0.4.3
     - importlib_metadata >=3.6
     - keyring >=15.1
-    - pkginfo >=1.4.2
-    - python
+    - pkginfo >=1.8.1
+    - python >=3.6
     - readme_renderer >=21.0
     - requests >=2.20
     - requests-toolbelt >=0.8.0,!=0.9.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:35]: Please remove `setuptools` from `run`, it's not mentioned in the upstream source https://github.com/pypa/twine/blob/d82cd00d216f6a1768ee835d1b534f16190f35bd/setup.cfg#L38

File: recipe/meta.yaml [hunk 3]
```diff
@@ -18,15 +19,16 @@ build:
 requirements:
   host:
     - pip
-    - python
+    - python >=3.6
     - setuptools >=0.7.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: setuptools should be >=45, see https://github.com/pypa/twine/blob/d82cd00d216f6a1768ee835d1b534f16190f35bd/pyproject.toml#L3

File: recipe/meta.yaml [hunk 4]
```diff
@@ -18,15 +19,16 @@ build:
...
     - setuptools_scm >=1.15
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: `setuptools_csm` should be `>=6.0` and add `toml`

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**